kong/plugins/http-log/handler.lua

```diff
@@ -189,12 +193,43 @@ function HttpLogHandler:log(conf)
 
   local queue_conf = Queue.get_plugin_params("http-log", conf, make_queue_name(conf))
   kong.log.debug("Queue name automatically configured based on configuration parameters to: ", queue_conf.name)
+  local entry = cjson.encode(kong.log.serialize())
+
+  if queue_conf.max_batch_size == 1 then
+    local ok ,err = timer_at(0, function()
```

Comment:
could we extract the timer's logic in a local function? Also - this looks very similar to the `process_once` method of the queue. Perhaps we could expose that logic from there and reuse it, to reduce duplication.

Reply:
I like the idea of function reuse. So the Queue has to expose a function(e.g. `Queue:handle`)  to be used internally and externally. 

Here is the function reuse approach PR https://github.com/Kong/kong/pull/13384. However, we have to refactor the Queue a little to expose 
- Queue.create(): create a Queue instance without starting a background timer, this helps us to leverage the OOP of the queue.
- Queue:handle(): a function contains retry logic

Reply:
I left a review comment, I hope my observation above didn't cost you too much time and efforts :smile_cat: